webapp/src/components/confirmationDialogBox.tsx

```diff
@@ -1,56 +1,67 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react'
+import React, {useCallback} from 'react'
 import {FormattedMessage} from 'react-intl'
 
 import Button from '../widgets/buttons/button'
 
 import Dialog from './dialog'
 import './confirmationDialogBox.scss'
 
+type ConfirmationDialogBoxProps = {
+    heading: string
+    subText?: string
+    confirmButtonText?: string
+    onConfirm: () => void
+    onClose: () => void
+}
+
 type Props = {
-    propertyId: string;
-    onClose: () => void;
-    onConfirm: () => void;
-    heading: string;
-    subText?: string;
+    dialogBox: ConfirmationDialogBoxProps
 }
 
 export const ConfirmationDialogBox = (props: Props) => {
+    const handleOnClose = useCallback(props.dialogBox.onClose, [])
```

Comment:
there is not reason to make the useCallback here, the `props.dialogBox.onClose` it is already a reference to a function, what makes sense is to use the `useCallback in the `carDetailProperties.tsx` wherever that function is created.

Reply:
Actually, I think we can just remove the usage of `useCallback here and let it not memoize the hander in this case, at the end of the day, is not so common to have this thing rendered, so I think it is ok to not memoize it.

Reply:
Removed useCallback in next commit.